archive/import.go

```diff
@@ -326,51 +319,62 @@ func (cw *chunkWriter) abort() {
 	os.Remove(cw.indexTempPath)
 }
 
-func (cw *chunkWriter) close(ctx context.Context) error {
+func (cw *chunkWriter) close(ctx context.Context) (Chunk, error) {
+	return cw.closeWithTs(ctx, cw.firstTs, cw.lastTs)
+}
+
+func (cw *chunkWriter) closeWithTs(ctx context.Context, firstTs, lastTs nano.Ts) (Chunk, error) {
 	err := cw.dataFileWriter.Close()
 	if closeErr := cw.indexTempWriter.Close(); err == nil {
 		err = closeErr
 	}
 	if err != nil {
-		return err
+		return Chunk{}, err
 	}
-	err = writeChunkMetadata(ctx, chunkMetadataPath(cw.ark, cw.tsDir, cw.chunk.Id), chunkMetadata{
-		First:       cw.chunk.First,
-		Last:        cw.chunk.Last,
-		Kind:        FileKindData,
-		RecordCount: cw.chunk.RecordCount,
-	})
+	chunkMd := chunkMetadata{
+		First:       firstTs,
+		Last:        lastTs,
+		Kind:        cw.kind,
+		RecordCount: cw.count,
+		Masks:       cw.masks,
+	}
+	err = writeChunkMetadata(ctx, chunkMetadataPath(cw.ark, cw.tsd, cw.id), chunkMd)
 	if err != nil {
-		return err
+		return Chunk{}, err
 	}
 	// Write the time seek index into the archive, feeding it the key/offset
 	// records written to indexTempPath.
 	tf, err := fs.Open(cw.indexTempPath)
 	if err != nil {
-		return err
+		return Chunk{}, err
 	}
 	defer func() {
 		tf.Close()
 		os.Remove(tf.Name())
 	}()
 	zctx := resolver.NewContext()
 	tfr := zngio.NewReader(tf, zctx)
-	idxURI := cw.chunk.seekIndexPath(cw.ark)
+	chunk := chunkMd.Chunk(cw.id)
```

Comment:
It looks like this close method writes one extra chunk based on the metadata, which will be an addition to the all the chunks written before calling close. Do I have this right, or am I missing something?

Reply:
A single instance of a `chunkWriter` will only write a single chunk. The close() method returns the `Chunk` that it wrote. There's a lot of book keeping & other calls happening, due mostly to writing the seek index file (as a temporary, the flushed during close) at the same time as writing the chunk data file.